### PR TITLE
Added replicaof parameter

### LINKED ISSUE #401: Replace `slaveof` with `replicaof`
As of 2018, the term `slaveof` was aliased to and then later replaced with `replicaof`. This module should support that same terminology to make it easier for users to align settings between redis itself and the redis module.

Making such a change to replace the term would be backwards compatible. I am not sure if we can find a way to alias the parameter or if we should just cut the cord on backwards compatibility.

The references to change are listed [here](https://github.com/voxpupuli/puppet-redis/search?q=slaveof). We would of course need a quick set of directions for users to migrate to the new settings, as well.

### PR DESCRIPTION
**Pull request by kolaCZek**
Signed-off-by: Martin Kolací <martin.kolaci@lmc.eu>

#### Pull Request (PR) description
Added **replicaof** parameter in config

#### This Pull Request (PR) fixes the following issues
Fixes #401

### CODE REVIEW

File: REFERENCE.md [hunk 1]
```diff
@@ -819,6 +820,14 @@ Data type: `Optional[String[1]]`
...
 
 
+Default value: ``undef``
+
+##### <a name="replicaof"></a>`replicaof`
+
+Data type: `Optional[String[1]]`
+
+
+
```
Comment (smortex) [inline REFERENCE.md:830]: Description is missing. Either do not update REFERENCE.md in the PR (we update this file when releasing a new version) or regenerate it and update the PR.

**Review by smortex — COMMENTED** [review]
Looks good, just a minor inconsistency.

File: REFERENCE.md (hunk 1 above)
Comment (jonasbartho) [reply REFERENCE.md:830]: Hi, can this PR be approved soon? 
"slaveof" has been deprecated for a while now...

Comment (kenyon) [reply REFERENCE.md:830]: As @smortex said, REFERENCE.md needs to be regenerated.

**Comment by smortex** [comment]
@MindTooth thanks for the heads-up!

**Pull request closed without merging**